plugins/outputs/sql/sql.go

```diff
@@ -248,15 +251,36 @@ func (p *SQL) Write(metrics []telegraf.Metric) error {
 		}
 
 		sql := p.generateInsert(tablename, columns)
-		_, err := p.db.Exec(sql, values...)
 
-		if err != nil {
-			// check if insert error was caused by column mismatch
-			p.Log.Errorf("Error during insert: %v, %v", err, sql)
-			return err
+		switch p.Driver {
+		case "clickhouse":
+			// ClickHouse needs to batch inserts with prepared statements
+			tx, err := p.db.Begin()
+			if err != nil {
+				return fmt.Errorf(": %v", err)
+			}
+			stmt, err := tx.Prepare(sql)
+			if err != nil {
+				return fmt.Errorf("Error during prepare: %v", err)
```

Comment:
Errors in Golang [should start lowercase without trailing punctuation](https://github.com/golang/go/wiki/CodeReviewComments#error-strings). How about
```suggestion
				return fmt.Errorf("prepare failed: %v", err)
```
Same for the other errors below.

Reply:
I see, I didn't know about that. Thank you for the link!